core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -615,11 +655,10 @@ object ZSchedule {
   final def collectUntilM[A](f: A => UIO[Boolean]): Schedule[A, List[A]] = this.doUntilM(f).collectAll
 
   /**
-   * A new schedule derived from the specified schedule which adds the delay
-   * specified as output to the existing duration.
+   * A new schedule derived from the specified schedule which transforms the delays into effectful sleeps.
    */
-  final def delayed[R, A](s: ZSchedule[R, A, Duration]): ZSchedule[R, A, Duration] =
-    s.modifyDelay((b, d) => IO.succeed(b + d)).reconsider((_, step) => step.copy(finish = () => step.delay))
+  final def delayed[R, A](s: ZSchedule[R, A, Duration]): ZSchedule[R with Clock, A, Duration] =
```

Comment:
Better way: `def delayed[R <: Clock, ...]` It will infer better!

Reply:
👍